mathesar/api/serializers/shared_serializers.py

```diff
@@ -138,6 +139,11 @@ class DisplayOptionsMappingSerializer(
         MathesarTypeIdentifier.DURATION.value: DurationDisplayOptionSerializer,
     }
 
+    def run_validation(self, data=empty):
+        if data == {}:
+            data = None
```

Comment:
@silentninja not changing the JSON data of display_options from {} to None causes an error here: https://github.com/centerofci/mathesar/blob/d93863041808d5cb472f20d2dbd2bc9e5c5f0d6c/mathesar/api/serializers/shared_serializers.py#L57
This makes the serailizer `None` as no display_oprions get matched.

Reply:
Do you suggest we add validation for `display_options` in the `SimpleColumnSerializer` ?

Reply:
> not changing the JSON data of display_options from {} to None causes an error here:

If a type does not have any equivalent display_option serializer, then it does not accept any display options, so we can just store an empty object in such cases rather than throwing an exception.